### PR TITLE
RPE-1055: Bitrise Step GitHub script runner

### PR DESCRIPTION
**Pull request by eugenio79**
As the title implies, it's a new Bitrise step that allows to run a remote script, without checking out the whole repo.
It is very similar to [Remote Bash Script Runner Step](https://github.com/bitrise-steplib/steps-remote-script-runner) but it allows you to do a couple of things more:

1. You can use it with private repos
2. You can pass command line arguments

If you want to test it, use the instructions in the `README.md`.
[This PR on riderapp-ios](https://github.com/deliveroo/riderapp-ios/pull/3481) is currently showing an example usage.

This Bitrise step hasn't been published yet (even though this repo is currently public...but we can switch to private if we want to).

### CODE REVIEW

File: step.yml [hunk 1]
```diff
@@ -49,39 +21,49 @@ is_always_run: false
...
+      summary: The default value is "master" but if your script leaves in a specific git branch, you need to specify it.
+      is_expand: true
+      is_required: true
+  - command_line_arguments:
+    opts:
+      title: The arguments to pass to the remote script
+      summary: Each script has its own usage and expects a completely different set of command line arguments. E.g. if a script usage says `script.sh --key1 val1 --key2 val2`, just write `--key1 val1 --key2 val2`, omitting the name of the script.
+      is_expand: true
+      is_required: true
+  - command_line_arguments:
```
Comment (StevenBanksRoo) [inline step.yml]: Looks like this is duplicated above. Not sure if thats intentional.

File: step.yml [hunk 2]
```diff
@@ -49,39 +21,49 @@ is_always_run: false
...
+      title: The git branch
+      summary: The default value is "master" but if your script leaves in a specific git branch, you need to specify it.
+      is_expand: true
+      is_required: true
+  - command_line_arguments:
+    opts:
+      title: The arguments to pass to the remote script
+      summary: Each script has its own usage and expects a completely different set of command line arguments. E.g. if a script usage says `script.sh --key1 val1 --key2 val2`, just write `--key1 val1 --key2 val2`, omitting the name of the script.
+      is_expand: true
+      is_required: true
```
Comment (StevenBanksRoo) [inline step.yml]: Do we need to support scripts without arguments?

File: step.sh [hunk 3]
```diff
@@ -1,22 +1,50 @@
...
+
+curl -H "Authorization: token $GITHUB_TOKEN" \
+  -H 'Accept: application/vnd.github.v3.raw' \
+  -sSL "https://api.github.com/repos/$repository/contents/$script_path?ref=$branch" &> remote-script.sh
+chmod +x remote-script.sh
+
+echo "Running the remote script..."
+
+command="./remote-script.sh $command_line_arguments"
+bash -c "$command"
```
Comment (StevenBanksRoo) [inline step.sh:60]: Just checking here. Do you know if errors are propagated? I mean, if an error occurs inside the remote script, does the Bitrise step fail?

File: step.yml (hunk 1 above)
Comment (eugenio79) [reply step.yml]: Nope, not intentional. Thanks for noticing it, removing it

File: step.yml (hunk 2 above)
Comment (eugenio79) [reply step.yml]: Good point. I set `is_required` to `false` and also I tested it by removing the `command_line_arguments` input in the `cli-test` (with the `test-script.sh`) and it's still working correctly 💪

File: step.sh (hunk 3 above)
Comment (eugenio79) [reply step.sh:60]: Good point. Luckily yes, it seems like they are propagated.
I tested it by not passing any command line arguments to the remote `promon-upload-file.sh` and the Bitrise workflow failed with the correct message/exit code:

```
It relies on the presence of the following environment variables:
* $PROMON_USER
* $PROMON_PASSWORD

--local-file-path       The path to the local file to upload
--remote-folder-path    The remote folder path where to upload the file

Usage: ./remote-script.sh --local-file-path <file_path_to_upload> --remote-folder-path <remote_folder_path>

|                                                                              |
+---+---------------------------------------------------------------+----------+
| x | Step CLI Test (exit code: 1)                                  | 1.84 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: ...hub.com/deliveroo/bitrise-step-github-script-runner/issues |
| Source: https://github.com/deliveroo/bitrise-step-github-script-runner       |
+---+---------------------------------------------------------------+----------+


+------------------------------------------------------------------------------+
|                               bitrise summary                                |
+---+---------------------------------------------------------------+----------+
|   | title                                                         | time (s) |
+---+---------------------------------------------------------------+----------+
| x | Step CLI Test (exit code: 1)                                  | 1.84 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: ...hub.com/deliveroo/bitrise-step-github-script-runner/issues |
| Source: https://github.com/deliveroo/bitrise-step-github-script-runner       |
+---+---------------------------------------------------------------+----------+
| Total runtime: 1.84 sec                                                      |
+------------------------------------------------------------------------------+
```

**Review by StevenBanksRoo — APPROVED** [review]
👍  Looks good.

**Review by StevenBanksRoo — APPROVED** [review]

**Pull request merged**